### PR TITLE
Fix dataframe auto datatype for empty and 1D inputs

### LINKED ISSUE #13294: gr.Dataframe with datatype="auto" crashes on empty list or 1D numpy array
### Describe the bug

\`gr.Dataframe\` with \`datatype="auto"\` crashes with \`IndexError\` when given an empty list or a 1D numpy array as the value.

**Root cause:** In \`gradio/components/dataframe.py\`, the \`set_auto_datatype\` method has two crash paths:

**1. Empty list (line 583):**
```python
elif isinstance(value, list):
    self.datatype = [... for val in value[0]]  # IndexError: list index out of range
```

**2. 1D numpy array (lines 568, 572):**
```python
elif isinstance(value, np.ndarray):
    self.datatype = [
        ... str(type(value[0, i]).__name__) ...  # IndexError: too many indices for array
        for i in range(value.shape[1])            # IndexError: tuple index out of range
    ]
```

A 1D array doesn't support 2D indexing (\`value[0, i]\`), and \`value.shape[1]\` fails because a 1D shape tuple has no second element.

### Have you searched existing issues?

Yes — no existing issue covers this.

### Reproduction

```python
import gradio as gr
import numpy as np

# Crash 1: empty list
gr.Dataframe(value=[], datatype="auto")
# IndexError: list index out of range

# Crash 2: 1D numpy array
gr.Dataframe(value=np.array([1, 2, 3]), datatype="auto")
# IndexError: tuple index out of range
```

### Expected behavior

Both cases should fall back gracefully to the default \`"str"\` datatype instead of crashing. The component already has an \`is_empty\` method that properly handles empty inputs — \`set_auto_datatype\` should use similar guards.

Suggested fix:

```python
elif isinstance(value, np.ndarray):
    if value.ndim < 2 or value.size == 0:
        self.datatype = "str"
        return
    # ... existing logic

elif isinstance(value, list):
    if len(value) == 0:
        self.datatype = "str"
        return
    # ... existing logic
```

### System info

Reproduced on \`main\` branch, \`gradio/components/dataframe.py\` lines 564-584.

### PR DESCRIPTION
**Pull request by fengfeng-zi**
## Summary
- guard `set_auto_datatype()` against empty lists and 1D numpy arrays
- fall back to `"str"` instead of indexing into unsupported shapes
- add regression tests covering both empty list and 1D ndarray inputs

Closes #13294

## Testing
- `python -m pytest test/components/test_dataframe.py -k "auto_datatype_handles_empty_list_and_1d_ndarray or is_empty"`

---

> [!NOTE]
> **Low Risk**
> Low risk: adds simple guards in `set_auto_datatype()` to avoid indexing errors on empty/1D inputs, plus focused regression tests.
> 
> **Overview**
> Fixes `gr.Dataframe(datatype="auto")` inference for unsupported/empty shapes by treating **empty lists** and **1D/empty numpy arrays** as a single `"str"` datatype instead of indexing into missing columns.
> 
> Adds regression tests ensuring `datatype` falls back to `"str"` for these inputs.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 3c565e800bbc0d38cc7fc8a8b01f8698c0c7c107. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: gradio/components/dataframe.py [hunk 1]
```diff
@@ -573,6 +576,9 @@ def set_auto_datatype(self, value):
             ]
 
         elif isinstance(value, list):
+            if len(value) == 0:
+                self.datatype = "str"
+                return
```
[the remark this answers is not in the document]
Comment (freddyaboulton) [reply gradio/components/dataframe.py:581]: Please fix @fengfeng-zi
